### DISCUSSION TITLE
v9 KubeOps Generator cant handle separate projects?

### DISCUSSION THREAD

**Discussion by ewassef** [dotnet/dotnet-operator-sdk#750] [Q&A]
We have always carved out our entities into a separate project and distributed them via Nuget for other projects to be able to create CRDs as needed, however, it seems that with the new codegen in the KubeOps.Generator class, it will fail to detect the entities (and everything else) if its only referenced in the operator (aspnet) project. But the weirder thing is that it will create multiple default classes in each of the referenced projects (creating duplicate methods and failing to compile) if its referenced in the entities project. 

Is this the desired behavior? whats the intent for the code generators? just to avoid reflection on startup?

**Comment by ewassef** [comment]
@buehler , run into this before?

**Reply by buehler** [reply]
Hey @ewassef 

Not until now :)

The generator only creates convenience methods for entities. As such, there should not be any problems. Maybe I misconfigured the generator? Was the first time I wrote a c# source generator.

**Reply by ewassef** [reply]
I've never done it either :) 
But I'm thinking that it might be as quick as adding a unique identifier to the filename. 
```csharp
context.AddSource(
            $"EntityDefinitions.{Guid.NewGuid()}.g.cs",
            SourceText.From(declaration.ToString(), Encoding.UTF8, SourceHashAlgorithm.Sha256));
```

Ill try it out and lyk

**Reply by ewassef** [reply]
looks like that would work https://github.com/buehler/dotnet-operator-sdk/pull/758

**Reply by buehler** [reply]
Hmm :-) nice

And the generator does not create conflicts with this? I thought that if new files are added, old ones may conflict.

**Reply by ewassef** [reply]
mmm. interesting.. ill test that specifically.. the tests that are there ran fine and didn't seem to mind
